app/utils/cache.py

```diff
@@ -1,27 +1,78 @@
 """
 Cache utilities for managing local file caching.
+
+Implements efficient caching for downloaded BERDLTables SQLite databases
+with age-based expiration and cleanup.
+
+Cache Structure:
+    {CACHE_DIR}/
+        {berdl_table_id}/
+            {pangenome_id}.db      # SQLite database
+            metadata.json          # Cache metadata (timestamps, checksums)
 """
 
+import json
+import time
+import shutil
+import logging
 from pathlib import Path
-from typing import Tuple
+from typing import Tuple, Optional, Dict, Any, List
```

Comment:
You don't need to use Optional, Dict, or List as those are built in python types